src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -0,0 +1,212 @@
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.ar.ArConstants;
+import org.kuali.kfs.module.ar.ArKeyConstants;
+import org.kuali.kfs.module.ar.ArPropertyConstants;
+import org.kuali.kfs.module.ar.businessobject.ContractsGrantsInvoiceDetail;
+import org.kuali.kfs.module.ar.businessobject.SystemInformation;
+import org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument;
+import org.kuali.kfs.module.ar.document.service.impl.ContractsGrantsInvoiceDocumentServiceImpl;
+import org.kuali.kfs.module.ar.report.PdfFormattingMap;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.kfs.sys.util.FallbackMap;
+import org.kuali.kfs.sys.util.ReflectionMap;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class CuContractsGrantsInvoiceDocumentServiceImpl extends ContractsGrantsInvoiceDocumentServiceImpl {
+
+    // Customized to default final status date to last modified date.
+    @Override
+    protected Map<String, String> getTemplateParameterList(ContractsGrantsInvoiceDocument document) {
+        ContractsAndGrantsBillingAward award = document.getInvoiceGeneralDetail().getAward();
+
+        Map cinvDocMap = new ReflectionMap(document);
+        Map<String, Object> parameterMap = new FallbackMap<String, Object>(cinvDocMap);
+
+        Map<String, Object> primaryKeys = new HashMap<>();
+        primaryKeys.put(KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR, document.getAccountingPeriod().getUniversityFiscalYear());
+        primaryKeys.put(KFSPropertyConstants.PROCESSING_CHART_OF_ACCT_CD, document.getAccountsReceivableDocumentHeader().getProcessingChartOfAccountCode());
+        primaryKeys.put(KFSPropertyConstants.PROCESSING_ORGANIZATION_CODE, document.getAccountsReceivableDocumentHeader().getProcessingOrganizationCode());
+        SystemInformation sysInfo = businessObjectService.findByPrimaryKey(SystemInformation.class, primaryKeys);
+        parameterMap.put(KFSPropertyConstants.DOCUMENT_NUMBER, document.getDocumentNumber());
+        if (ObjectUtils.isNotNull(document.getDocumentHeader().getWorkflowDocument().getDateCreated())) {
+            parameterMap.put(KFSPropertyConstants.DATE, getDateTimeService().toDateString(document.getDocumentHeader().getWorkflowDocument().getDateCreated().toDate()));
+        }
+        if (ObjectUtils.isNotNull(document.getDocumentHeader().getWorkflowDocument().getDateFinalized())) {
+            parameterMap.put(ArPropertyConstants.FINAL_STATUS_DATE, getDateTimeService().toDateString(document.getDocumentHeader().getWorkflowDocument().getDateFinalized().toDate()));
+        }
+        else if (ObjectUtils.isNotNull(document.getDocumentHeader().getWorkflowDocument().getDateLastModified())) {
+            parameterMap.put(ArPropertyConstants.FINAL_STATUS_DATE, getDateTimeService().toDateString(document.getDocumentHeader().getWorkflowDocument().getDateLastModified().toDate()));
```

Comment:
It looks like this part of the function is the only thing we are customizing here.  I suggest we create a new function to determine and set the final_status_date.  This will make it easier to identify the CU custom code in the future.
Also, you may want to use the source code formatter in eclipse, the else if link should be on the same as the related If statement's closing brace